src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -34,207 +44,177 @@ internal sealed class SlidingTextWindow : IDisposable
 
         /// <summary>
-        /// Returns the start of the current lexeme relative to the window start.
+        /// Similar to <see cref="_characterWindowStartPositionInText"/>, except this represents the index (exclusive) of the first character
+        /// that <see cref="_characterWindow"/> encompases in <see cref="Text"/>.  This is equal to <see cref="_characterWindowStartPositionInText"/>
+        /// + <see cref="_characterWindow"/>'s <see cref="ArraySegment{T}.Count"/>.
         /// </summary>
-        public int LexemeRelativeStart
-        {
-            get
-            {
-                return _lexemeStart;
-            }
-        }
+        private readonly int CharacterWindowEndPositionInText => _characterWindowStartPositionInText + _characterWindow.Count;
 
         /// <summary>
-        /// Number of characters in the character window.
+        /// Returns true if <paramref name="position"/> is within the current character window, and thus the character at that position
+        /// can be read from the character window without going back to the underlying <see cref="Text"/>.
         /// </summary>
-        public int CharacterWindowCount
+        private readonly bool PositionIsWithinWindow(int position)
         {
-            get
-            {
-                return _characterWindowCount;
-            }
+            return position >= _characterWindowStartPositionInText &&
+                   position < this.CharacterWindowEndPositionInText;
         }
 
         /// <summary>
-        /// The absolute position of the start of the current lexeme in the given
-        /// SourceText.
+        /// Returns true if <paramref name="span"/> is within the current character window, and thus the sub-string corresponding to 
+        /// that span can be read can be read from the character window without going back to the underlying <see cref="Text"/>.
         /// </summary>
-        public int LexemeStartPosition
+        public readonly bool SpanIsWithinWindow(TextSpan span)
         {
-            get
-            {
-                return _basis + _lexemeStart;
-            }
+            return span.Start >= _characterWindowStartPositionInText &&
+                   span.End <= this.CharacterWindowEndPositionInText;
         }
 
         /// <summary>
-        /// The number of characters in the current lexeme.
+        /// Returns the span of characters corresponding to <paramref name="span"/> from <see cref="Text"/> <em>if</em>
+        /// <paramref name="span"/> is entirely within bounds of the current character window (see <see
+        /// cref="SpanIsWithinWindow(TextSpan)"/>).  Otherwise, returns <see langword="false"/>.  Used to allow
+        /// fast path access to a sequence of characters in the common case where they are in the window, while
+        /// falling back if they are not.
```

Comment:
There's no fallback in this method, just returning default. Perhaps "while not incurring read costs if not in the window"?